### DISCUSSION TITLE
Spawnaction not spawning on Mainpath properly

### DISCUSSION THREAD

**Discussion by ubertpendragon** [FakeFishGames/Barotrauma#14140] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

In events, if mainpath is specified in Spawnaction, the spawned entity will sometimes be spawned in the abyss or in a cave.

Tested with both SpawnLocation and SpawnLocationType

`triggerevent mainpathspawntest`
```xml    
    <ScriptedEvent identifier="mainpathspawntest">
      <SpawnAction speciesname="crawler" amount="3" SpawnLocation="MainPath" offset="500" />
    </ScriptedEvent>
```

![image](https://github.com/FakeFishGames/Barotrauma/assets/73229309/38ef13f4-0ff0-4e08-8e73-0520d7e43bd2)
![image](https://github.com/FakeFishGames/Barotrauma/assets/73229309/83559b44-a3ee-43c3-895b-9d1ceb837cd6)
![image](https://github.com/FakeFishGames/Barotrauma/assets/73229309/1d3a2bce-4cd1-44c8-875d-607678f63b95)

### Reproduction steps

.

### Bug prevalence

Just once

### Single player or multiplayer?

Single player

### Version

v1.4.6.0 (Blood in the Water Update, hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
Thank you for the report! This has now been fixed in our private repo (https://github.com/FakeFishGames/Barotrauma-development/commit/14fc3242040533cf97a5e32ad08d3a27a579f93e), closing.
